returnn/tf/layers/basic.py

```diff
@@ -1341,6 +1341,19 @@ def __init__(self, position, axis, **kwargs):
     # (BatchAxes.., InputAxesBeforeGatherAxis, PositionAxes.., InputAxesAfterGatherAxis..)
     self.output.placeholder = tf.gather(params=params, indices=indices, axis=gather_axis, batch_dims=batch_dims)
 
+    if input_data.dim_tags[old_gather_axis].is_batch_dim():
+      for dim_tag in self.output.dim_tags:
+        if dim_tag.is_spatial_dim():
+          axis = self.output.get_batch_axis_excluding_batch(self.output.get_axis_by_tag_name(dim_tag.description))
```

Comment:
This is:

* way too complicated: you can simply do `for axis, dim_tag in enumerate(self.output.dim_tags)`
* wrong: do not rely on `get_axis_by_tag_name` and `dim_tag.description`
* not necessary: just use `dim_tag.dyn_size_ext`